src/pages/home/report/ReportActionContextMenu.js

```diff
@@ -89,34 +29,115 @@ const propTypes = {
 
     // Controls the visibility of this component.
     isVisible: PropTypes.bool,
+
+    // The session of the logged in person
+    session: PropTypes.shape({
+        // Email of the logged in person
+        email: PropTypes.string,
+    }),
 };
 
 const defaultProps = {
     reportAction: {},
     isMini: false,
     isVisible: false,
+    session: {},
 };
 
-const ReportActionContextMenu = (props) => {
-    const wrapperStyle = getReportActionContextMenuStyles(props.isMini);
-    return props.isVisible && (
-        <View style={wrapperStyle}>
-            {CONTEXT_ACTIONS.map(contextAction => contextAction.shouldShow && (
-                <ReportActionContextMenuItem
-                    icon={contextAction.icon}
-                    text={contextAction.text}
-                    successIcon={contextAction.successIcon}
-                    successText={contextAction.successText}
-                    isMini={props.isMini}
-                    onPress={() => contextAction.onPress(props.reportAction)}
-                    key={contextAction.text}
-                />
-            ))}
-        </View>
-    );
-};
+class ReportActionContextMenu extends React.Component {
+    constructor(props) {
+        super(props);
+
+        /**
+         * A list of all the context actions in this menu.
+         */
+        this.contextActions = [
```

Comment:
Why are these being moved inside here?

Reply:
I kind of think that this should be set in the state and the view shouldn't display anything until these are set in the state (with an early return in the render method). I suggest something like this:

1. All callbacks are changed to class methods and bound in the constructor
2. `contextActions` is moved to the state and starts out empty
3. In `componentDidMount()` use `this.props` to properly construct the `contextActions` and set them in the state

Reply:
> Why are these being moved inside here

So the callbacks can access the component's props.

Reply:
are you happy with this @tgolen?

Reply:
I think this is better yeah, but still not quite there yet. What I think we settled on in the other thread is:

1. Move `this.contextActions` to be back at the top of the file outside of the class
2. Change `onPress={contextAction.onPress}` to be `onPress={() => contextAction.onPress(this.props)}`
3. Keep the callbacks defined as part of the `contextActions` object (no need to declare them as separate functions).

Reply:
I'm confused by what you mean with 
> 3. Keep the callbacks defined as part of the contextActions object (no need to declare them as separate functions).

Do you mean getting back the logic to coppy to clipboard as part of the onPress, instead of a function outside the class and `contextActions`

Reply:
Hm, all I meant with that point is that it should look like this:
```
contextActions = [{
   onPress: (props) => {
        // do wahtever
   },
}];
```

Instead of like this:

```
function doWhatever(props) {
    // do whatever
}
contextActions = [{
   onPress: doWhatever,
}];
```

Reply:
It seems like this was never resolved. Was there a reason it wasn't?

Reply:
I did do the change, like you requested, but this was reverted again when I merged main in, and it had the changes for `edit comments` where `ReportActionContextMenu` had been set as a class. I decided to leave as it was in `main` but happy to go back and `move this.contextActions to be back at the top of the file outside of the class`

Reply:
OK, thanks. I still think it would be good to move these back to be outside the class 👍

Reply:
sooooo, heh, sorry to keep commenting here, but are these going to be moved outside the class?

Reply:
sorry, I can definetly do it, I'm a bit apreensive on doing that change, as I'm not sure if I'll be able to maintain some of the comments you and Rory have asked me to do. but let me give it a go.

Reply:
I've done some changes to go in this direction, but I'm not entirely sure about them, feel there should be a better way to do this, there are so many calls to `this` and `props` that I ended up setting it as a method receiving the context menu. Let me know your thoughts.